source/mobile/build-custom-mobile-apps.rst

```diff
@@ -0,0 +1,89 @@
+- Customers using self-hosted Mattermost Team Edition or Enterprise Editions that prefer to host their own push proxy server instead of using one of Mattermost’s hosted versions.
+  
+Deployment Options
+------------------
+
+When you decide to build your own Mattermost Mobile apps, you have multiple ways to deploy: 
+
+- Submitting your app to `an Enterprise App Store <#using-an-enterprise-app-store>`_.
+- Using `an Enterprise Mobile Management (EMM) provider <#using-an-emm-provider>`_.
+- Submitting your app to `public app stores <#using-public-app-stores>`_.
+- Using `another distribution method <#using-another-distribution-method>`_.
+
+Developing and Maintaining Your Custom App
+------------------------------------------
+
+The Mattermost Mobile App is an open source project. Customizing Mattermost Mobile apps requires a fork of the source code. Your team will be responsible for maintaining that fork, as well as keeping that fork updated with any changes made by Mattermost.
+
+This means that you manage the maintenance of your custom Mattermost Mobile apps, such as rebuilding and incorporating feature and/or security updates. If this isn't done regularly, your applications won't match the functionality of our publicly-available applications, and could be incompatible with future versions of Mattermost Server.
+
+This process can be complicated and can greatly increase deployment time, not only initially, but whenever the mobile apps need to be updated. We recommend having your development team `review the Mattermost Mobile Apps developer documentation <https://developers.mattermost.com/contribute/mobile/>`__ to ensure they understand the scale and requirements of taking this path. This documentation provides guidance on building, compiling, signing, and white-labeling Mattermost Mobile apps.
+
+In general, building your own Mobile apps will present some challenges, including:
+
+- Installing the necessary developer tools (such as Nodejs, XCode Developer Tools, Android SDKs, as well as others).
+- Obtaining and providing certificates for your custom Mattermost Mobile apps*.
+- Signing your custom Mattermost Mobile apps*.
+- Distributing your Mobile app to your users.
+
+***** Mattermost Mobile apps are signed, and they have certificates and keys associated with Mattermost and public app stores. This means Mattermost’s Mobile App won’t work if you choose to privately host the Mattermost Push Proxy Service (MPNS). You'll need to build your own custom mobile app.
+
+Custom Mobile App Branding
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ensure you select a unique app name that helps users distinguish your version from others, such as "<Your Company Name> Chat". See our `Brand and Visual Design Guidelines <https://handbook.mattermost.com/operations/operations/company-processes/publishing/publishing-guidelines/brand-and-visual-design-guidelines#name-usage-guidelines.html>`__ in our Company Handbook for details.
```

Comment:
> <Your Company Name> Chat

do we want to call it "Chat" or would it be better as "Messaging"?

Reply:
Messaging is likely a better choice. Thanks!